src/test/components/FlameGraph.test.js

```diff
@@ -5,66 +5,114 @@
+    const {
+      profile,
+      funcNamesPerThread: [funcNames],
+    } = getProfileFromTextSamples(`
     A[cat:DOM]       A[cat:DOM]       A[cat:DOM]
     B[cat:DOM]       B[cat:DOM]       B[cat:DOM]
     C[cat:Graphics]  C[cat:Graphics]  H[cat:Network]
     D[cat:Graphics]  F[cat:Graphics]  I[cat:Network]
     E[cat:Graphics]  G[cat:Graphics]
   `);
 
-  const store = storeWithProfile(profile);
+    const store = storeWithProfile(profile);
+    const { getState, dispatch } = store;
 
-  const { container } = render(
-    <Provider store={store}>
-      <FlameGraph />
-    </Provider>
-  );
+    const { container } = render(
+      <Provider store={store}>
+        <FlameGraph />
+      </Provider>
+    );
 
-  flushRafCalls();
+    flushRafCalls();
 
-  const drawCalls = ctx.__flushDrawLog();
+    const getContentDiv = () =>
+      ensureExists(
+        container.querySelector('.flameGraphContent'),
+        `Couldn't find the content div with selector .flameGraphContent`
+      );
 
-  expect(container.firstChild).toMatchSnapshot();
-  expect(drawCalls).toMatchSnapshot();
+    return {
+      getState,
+      dispatch,
+      container,
+      ctx,
+      getContentDiv,
+      funcNames,
+    };
+  }
 
-  delete window.devicePixelRatio;
-});
+  it('matches the snapshot', () => {
+    const { container, ctx } = setup();
+    const drawCalls = ctx.__flushDrawLog();
+    expect(container.firstChild).toMatchSnapshot();
+    expect(drawCalls).toMatchSnapshot();
+  });
 
-it('renders a message instead of FlameGraph when call stack is inverted', () => {
-  const { profile } = getProfileFromTextSamples(`
-    A  B
-  `);
+  it('renders a message instead when call stack is inverted', () => {
+    const { container, dispatch } = setup();
+
+    dispatch(changeInvertCallstack(true));
+
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  it('can be navigated with the keyboard', () => {
+    const { getState, dispatch, getContentDiv, funcNames } = setup();
+    const div = getContentDiv();
+
+    function selectedNode() {
+      const callNodeIndex = selectedThreadSelectors.getSelectedCallNodeIndex(
+        getState()
+      );
+      return callNodeIndex && funcNames[callNodeIndex];
+    }
+
+    dispatch(changeSelectedCallNode(0, [0, 1] /* B */));
+
+    expect(selectedNode()).toBe('B');
+
+    fireEvent.keyDown(div, { key: 'ArrowUp' });
+
+    expect(selectedNode()).toBe('C');
+
+    fireEvent.keyDown(div, { key: 'ArrowRight' });
+
+    expect(selectedNode()).toBe('H');
+
+    fireEvent.keyDown(div, { key: 'ArrowLeft' });
 
-  const store = storeWithProfile(profile);
-  store.dispatch(changeInvertCallstack(true));
+    expect(selectedNode()).toBe('C');
 
-  const { container } = render(
-    <Provider store={store}>
-      <FlameGraph />
-    </Provider>
-  );
+    fireEvent.keyDown(div, { key: 'ArrowDown' });
 
-  expect(container.firstChild).toMatchSnapshot();
+    expect(selectedNode()).toBe('B');
+  });
```

Comment:
nit: I'd remove the empty lines for groups where this makes sense (eg: fireEvent something + the matching expectation). Also adding some comments for each of the "groups" explaining the expectations